src/vscode/devcontainer.ts

```diff
@@ -13,23 +13,53 @@ import { Task } from "../task";
  */
 const DEVCONTAINER_FILE = ".devcontainer.json";
 
+/**
+ * devcontainer features options
+ */
+export class DevEnvironmentFeature {
+  public readonly featureName?: string;
+  public readonly version?: string;
+
+  private constructor() {}
+}
```

Comment:
This should just be an interface.
Also I assume the `featureName` is always required?
And since it is only for dev containers rename to `DevContainerFeature`
```suggestion
export interface DevContainerFeature {
  public readonly featureName: string;
  public readonly version?: string;
}
```